### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/earthly/commit/781a7b04651ce7c93b9852f1db57de28b2ab2038#commitcomment-88328415

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/earthly/commit/6ab0715cde12f6fa99922de2b27349fca9026c2b

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
 - Support for all ssh-based key types (e.g. ssh-ed25519), and not only ssh-rsa. [#1783](https://github.com/earthly/earthly/issues/1783)
 
 ### Fixed
-- Unable to specify public key to add via the command-line, e.g. running `earthly account add-key <key>` ignored the key and fellback to an interactive prompt.
+- Unable to specify public key to add via the command-line, e.g. running `earthly account add-key <key>` ignored the key and fell back to an interactive prompt.
```
Comment (jsoref) [inline CHANGELOG.md:26]: Some projects don't like changing changelogs. Happy to drop this or anything else.

File: builder/builder.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ type Opt struct {
 	Console                               conslogging.ConsoleLogger
 	Verbose                               bool
 	Attachables                           []session.Attachable
-	Enttlmnts                             []entitlements.Entitlement
+	Entitlements                          []entitlements.Entitlement
```
Comment (jsoref) [inline builder/builder.go:58]: Sometimes capital letters hint at public apis that can't be changed. In general, assuming this isn't a public api, there didn't seem to be a reason not to use the full word, especially give that it's only one character longer than `Attachables`...

File: builder/builder.go [hunk 3]
```diff
@@ -211,7 +211,7 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 		}
 
 		// WARNING: the code below is deprecated, and will eventually be removed, in favour of wait_block.go
-		// This code is only used when dealing with VERISON 0.5 and 0.6; once these reach end-of-life, we can
+		// This code is only used when dealing with VERSION 0.5 and 0.6; once these reach end-of-life, we can
```
Comment (jsoref) [inline builder/builder.go:214]: Google Sheets thinks this should be Verizon...

File: cloud/account.go [hunk 4]
```diff
@@ -264,7 +264,7 @@ func (c *client) ping(ctx context.Context) (email string, writeAccess bool, err
 	return resp.Email, resp.WriteAccess, nil
 }
 
-func (c *client) AccountResetReqestToken(ctx context.Context, email string) error {
+func (c *client) AccountResetRequestToken(ctx context.Context, email string) error {
```
Comment (jsoref) [inline cloud/account.go:267]: It's possible this is a public function...

File: cloud/auth.go [hunk 5]
```diff
@@ -388,7 +388,7 @@ func (c *client) saveToken() error {
 	return nil
 }
 
-func (c *client) loginWithPassowrd(ctx context.Context) error {
+func (c *client) loginWithPassword(ctx context.Context) error {
```
Comment (jsoref) [inline cloud/auth.go:391]: Sometimes projects try to be cute, especially when writing passw0rds, but this didn't seem like such a case...

File: config/config.go [hunk 6]
```diff
@@ -64,7 +64,7 @@ type GlobalConfig struct {
 	BuildkitRestartTimeoutS  int      `yaml:"buildkit_restart_timeout_s" help:"How long to wait for buildkit to (re)start, in seconds."`
 	BuildkitAdditionalArgs   []string `yaml:"buildkit_additional_args"   help:"Additional args to pass to buildkit when it starts. Useful for custom/self-signed certs, or user namespace complications."`
 	BuildkitAdditionalConfig string   `yaml:"buildkit_additional_config" help:"Additional config to use when starting the buildkit container; like using custom/self-signed certificates."`
-	BuildkitMaxParallelism   int      `yaml:"buildkit_max_parallelism"   help:"Max parallelism for builtkit workers"`
+	BuildkitMaxParallelism   int      `yaml:"buildkit_max_parallelism"   help:"Max parallelism for buildkit workers"`
```
Comment (jsoref) [inline config/config.go:67]: brand

File: docs/basics/part-1-a-simple-earthfile.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ docker:
 
 Throughout this tutorial, we'll build up this example Earthfile from scratch and then add even more to it. By the end you'll have a better grasp of how Earthly works and the power and repeatability it can bring to your build process.
 
-This tutorial focuses on using Earthly with a Go project, but you can find examples of Earthfiles for [Python](#more-examples), [Javascript](#more-examples) and [Java](#more-examples) at the bottom of each page.
+This tutorial focuses on using Earthly with a Go project, but you can find examples of Earthfiles for [Python](#more-examples), [JavaScript](#more-examples) and [Java](#more-examples) at the bottom of each page.
```
Comment (jsoref) [inline docs/basics/part-1-a-simple-earthfile.md:21]: brand

File: docs/ci-integration/guides/codebuild-integration.md [hunk 8]
```diff
@@ -21,7 +21,7 @@ phases:
       - wget https://github.com/earthly/earthly/releases/download/v0.6.28/earthly-linux-amd64 -O /usr/local/bin/earthly && chmod +x /usr/local/bin/earthly
   pre_build:
     commands:
-      - echo Logging in to Docker
+      - echo Logging into Docker
```
Comment (jsoref) [inline docs/ci-integration/guides/codebuild-integration.md:24]: This correction is my newest and has been the most controversial of late. Happy to drop.

File: docs/ci-integration/guides/google-cloud-build.md [hunk 9]
```diff
@@ -112,7 +112,7 @@ Running this build will use the [`cloudbuild.yaml`](https://github.com/earthly/c
       - +docker
 ```
 
-[The second step](hhttps://github.com/earthly/ci-example-project/blob/ea44992b020b52cb5a46920d5d11d4b8389ce19d/cloudbuild.yaml#L8-L13) runs a sample, Google Cloud Build only example to show how you would use an external service account to do things that normally requires credentials.
```
Comment (jsoref) [inline docs/ci-integration/guides/google-cloud-build.md:115]: broken link

File: earthfile2llb/interpreter.go [hunk 10]
```diff
@@ -891,19 +891,19 @@ func (i *Interpreter) handleCopy(ctx context.Context, cmd spec.Command) error {
 	for index, src := range srcs {
 		var artifactSrc domain.Artifact
 		var parseErr error
-		if isInParansForm(src) {
+		if isInParamsForm(src) {
 			// COPY (<src> <flag-args>) ...
-			artifactStr, extraArgs, err := parseParans(src)
+			artifactStr, extraArgs, err := parseParams(src)
```
Comment (jsoref) [inline earthfile2llb/interpreter.go:896]: I couldn't really tell if this was `parens` or `params`. I've taken a guess. I'm not claiming it's educated.

It's possible this is a term-of-art...

File: examples/go-monorepo/README.md [hunk 11]
```diff
@@ -6,7 +6,7 @@ The example contains two Microservices in a `services` directory, and a single s
 
 Each sub-module of this monorepo manages it's own build, test, and release steps, with parent Earthfile acting as an orchestrator.
 
-The `services` import code from `libs` during their Earthly build by utilizing [artifacts](https://docs.earthly.dev/docs/earthfile?q=save+artifct).
+The `services` import code from `libs` during their Earthly build by utilizing [artifacts](https://docs.earthly.dev/docs/earthfile?q=save+artifact).
```
Comment (jsoref) [inline examples/go-monorepo/README.md:9]: I hope this resource doesn't expect that spelling...

File: tests/remote-cache/test.sh [hunk 12]
```diff
@@ -28,7 +28,7 @@ export REGISTRY="$REGISTRY_IP:5000"
 
 if test -z "$REGISTRY_IP"
 then
-     echo echo "Error: REGISTRY_IP is empty"
+     echo "Error: REGISTRY_IP is empty"
```
Comment (jsoref) [inline tests/remote-cache/test.sh:31]: I've decided to guess that this is errant. Although, it's certainly possible it's constructing a `Dockerfile` where it wants a level of `echo ...`. If that's the case, I'd suggest wrapping the content w/ `'...'`

File: tests/save-artifact-legacy.earth [hunk 13]
```diff
@@ -8,13 +8,13 @@ savea:
...
 # second test: saveb is triggered by
-# a BUILD in savec, and savec is inherrited by
+# a BUILD in savec, and savec is inherited by
 # saved. when +saved is run, we should
 # produce files b, c, and d.
 
 saveb:
     FROM alpine:latest
-    RUN echo bannana > b
+    RUN echo banana > b
```
Comment (jsoref) [inline tests/save-artifact-legacy.earth:17]: I can't tell why this fruit was spelled this way.

`cherry` is spelled correctly.

I haven't fixed all spellings of `banana`...

File: tests/shell-out/old.earth [hunk 14]
```diff
@@ -50,36 +50,36 @@ test6:
 
 test7:
     FROM alpine:3.15
-    ARG varriety=Q2hhbXBpb24=
-    BUILD +b64decoder --mydata="$varriety" --pattern="Champion"
+    ARG variety=Q2hhbXBpb24=
+    BUILD +b64decoder --mydata="$variety" --pattern="Champion"
```
Comment (jsoref) [inline tests/shell-out/old.earth:54]: It's possible this is a term of art, or `var`+`riety`...

File: util/containerutil/settings_test.go [hunk 15]
```diff
@@ -328,7 +328,7 @@ func TestBuildArgMatrixValidationFailures(t *testing.T) {
 	}
 }
 
-func TestParseAndvalidateURLFailures(t *testing.T) {
+func TestParseAndValidateURLFailures(t *testing.T) {
```
Comment (jsoref) [inline util/containerutil/settings_test.go:331]: This change isn't necessary, strictly speaking. But given the style here, there didn't seem to be a good reason not to make it.

**Review by jsoref — COMMENTED** [review]
Most corrections suggested by Google Sheets. All fault mine.

**Comment by alexcb** [comment]
Thanks for the PR, it looks like some of these changes broke the build.

Take a look at https://github.com/earthly/earthly/blob/main/CONTRIBUTING.md#running-tests to see how to run our tests locally.

I'll gladly merge the PR if all our tests pass, thanks!

**Review by alexcb — APPROVED** [review]
thanks for all the fixes, and putting up with our flaky tests.

**Pull request merged**